CoEpi/ble/Peripheral.swift

```diff
@@ -107,34 +117,26 @@ extension PeripheralImpl: CBPeripheralManagerDelegate {
 
     func peripheralManager(_ peripheral: CBPeripheralManager, didReceiveRead request: CBATTRequest) {
         os_log("Peripheral manager did receive read request: %@", log: blePeripheralLog, request.description)
+
+        let respondClosure: (Data) -> () = { [weak self] (responseData) in
+            // do we need to capture request & peripheral?
+            let theirData: Data? = request.value
+            request.value = responseData
+            peripheral.respond(to: request, withResult: .success)
+
+            let contact: Contact2 = .init(theirData: theirData, ourData: responseData)
+            self?.addNewContactEvent(contact: contact)
```

Comment:
Do we have to do anything when someone reads our CEN?

Reply:
What do you mean? I mean we need to callback the consumer so that we can do stuff like the `DebugViewController`, monitoring what peripherals we've connected with

Reply:
I think that this currently is distracting. At this moment we don't plan to do anything with this event. Let's tailor the interface to the spec. Add a log if needed.

Reply:
Another topic, concerning the closure, does this work if it's an async/long running operation? It may depend only on the BLE timeout?

Reply:
The `DebugViewModel` does use it - `let peripheralContactSent = String(decoding: peripheralContactSent?.ourData ?? Data(), as: UTF8.self)`, because it displays when contacts are made. I realize its only for debugging, so we could remove this event and kill that functionality?

Reply:
My bad. Yes, we should leave it. But make it more explicitly for debugging `didReadCharacteristic`, `didReadCen` or similar.

Reply:
Current code is not decoding ourData properly, resulting in jumbled string.

Reply:
Renamed to `didReadCharacteristic`